Add actual date and time to 'timeago' dates (#5085)

darkHarry opened the pull request:
Fixes #5085.

As per @dstufft every date time is stored in UTC so I added "UTC" in the end. Let me know if that's not what I was supposed to do.
* Added a tooltip for date time to 'timeago' dates with timezone

di commented:
I think the right thing to do here would be to localize the datetime prior to using it, and then add `%Z` to the formatting string:

```python
>>> dt
datetime.datetime(2018, 12, 26, 13, 36, 5, 789013)
>>> import pytz
>>> pytz.utc.localize(dt)
datetime.datetime(2018, 12, 26, 13, 36, 5, 789013, tzinfo=<UTC>)
>>> _.strftime('%Y-%m-%d %H:%M %Z')
'2018-12-26 13:36 UTC'
```

darkHarry commented:
Hi @di! Thanks for the review. I know printing "UTC" was plain stupid :P 
I actually ran into a problem regarding the all_releases query object in /warehouse/packaging/views.py which passes the all_releases list to the template, as it can't be modified so I converted it to a dict but then 1 test failed and moreover, we can't import pytz in the template. Any help appreciated

di commented:
I'd recommend adding it [as a filter](https://github.com/pypa/warehouse/blob/1fbb4ac752e68b5840b9e09b68e44a165569bfa6/warehouse/filters.py) instead, so we could use it in the Jinja template whenever we need to:

```
{{ hversion.created.strftime('%Y-%m-%d %H:%M') | localize_datetime }}

darkHarry commented:
Thanks, @di!
I've added a filter to localize datetime and its corresponding unit test. 
Also, I found another filter to format datetime so, i removed the ``strftime()`` and used that instead.

darkHarry commented on warehouse/filters.py:
```diff
@@ -156,3 +157,7 @@ def parse_version(version_str):
+
+
+def localize_datetime(timestamp):
+    return pytz.utc.localize(timestamp)
```
sure

di commented:
Thanks @darkHarry!

darkHarry commented:
It was Awesome. Thanks a lot, @di.
